cmd/webhook/main.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"github.com/golang/glog"
+	"k8s.io/klog"
+	"net/http"
+	"os"
+	"os/signal"
+	"sigs.k8s.io/service-apis/pkg/admission"
+	"syscall"
+)
+
+var (
+	tlsCert, tlsKey string
+)
+
+func main() {
+	flag.StringVar(&tlsCert, "tlsCertFile", "/etc/certs/cert", "File with x509 certificate")
+	flag.StringVar(&tlsKey, "tlsKeyFile", "/etc/certs/key", "File with private key to tlsCertFile")
+
+	flag.Parse()
+
+	certs, err := tls.LoadX509KeyPair(tlsCert, tlsKey)
+	if err != nil {
+		klog.Fatalf("failed to load admission webhook keypair with err: %v", err)
+	}
+
+	server := &http.Server{
+		Addr:      ":8443",
+		TLSConfig: &tls.Config{Certificates: []tls.Certificate{certs}},
+	}
+	mux := http.NewServeMux()
+	mux.HandleFunc("/validate", admission.ServeHTTP)
+	server.Handler = mux
+
+	go func() {
+		err := server.ListenAndServeTLS("", "")
+		klog.Fatalf("admission webhook server stopped with err: %v", err)
+	}()
+
+	glog.Info("admission webhook server started and listening on :8443")
```

Comment:
Do you think we should have a command to run the server rather than starting the server from the binary?
I see at least two commands as being important: `server` and `version`. It should be possible to know the version of a binary easily.

Reply:
Ah I see so:
```
./binary # run the webhook server
./binary --version # print the version and exit(0)
```

That works as well.